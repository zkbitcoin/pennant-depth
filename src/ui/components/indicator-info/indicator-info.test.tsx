import React from "react";
import "@testing-library/jest-dom";
import { render, screen, act } from "@testing-library/react";
import { IndicatorInfo } from "./indicator-info";

describe("IndicatorInfo component", () => {
    const infoData = [
        { id: "index", label: "RSI", value: "100", displayWhileNoTrading: true },
    ];

    test("renders correctly with noTrading=false", () => {
        let container: HTMLElement;
        act(() => {
            ({ container } = render(<IndicatorInfo title="RSI" info={infoData} noTrading={false} />));
        });

        const wrapper = screen.getByTestId("indicator-info-wrapper");

        // Check that the flattened text content contains the correct strings
        expect(wrapper.textContent).toContain("RSI");
        expect(wrapper.textContent).toContain("100");
        expect(wrapper.textContent).not.toContain("No trading");

        expect(wrapper).toMatchSnapshot();
    });

    test("renders correctly with noTrading=true", () => {
        let container: HTMLElement;
        act(() => {
            ({ container } = render(<IndicatorInfo title="RSI" info={infoData} noTrading={true} />));
        });

        const wrapper = screen.getByTestId("indicator-info-wrapper");

        expect(wrapper.textContent).toContain("RSI");
        expect(wrapper.textContent).toContain("100");
        expect(wrapper.textContent).toContain("No trading");

        expect(wrapper).toMatchSnapshot();
    });

    test("renders close button if closeable=true", () => {
        const onCloseMock = jest.fn();
        act(() => {
            render(
                <IndicatorInfo
                    title="RSI"
                    info={infoData}
                    noTrading={false}
                    closeable={true}
                    onClose={onCloseMock}
                />
            );
        });

        const closeButton = screen.getByTitle("Remove");
        expect(closeButton).toBeInTheDocument();

        // simulate click
        closeButton.click();
        expect(onCloseMock).toHaveBeenCalled();
    });
});
