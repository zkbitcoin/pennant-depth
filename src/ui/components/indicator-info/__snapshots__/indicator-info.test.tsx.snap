// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorInfo component renders correctly with noTrading=false 1`] = `
<div
  class="indicator-info-wrapper"
  data-testid="indicator-info-wrapper"
>
  <span
    class="text-muted"
  >
    RSI: 
  </span>
  <div
    class="indicator-info__item"
  >
    <span
      class="text-muted"
    >
      RSI 
    </span>
    <span
      class="monospace-text"
    >
      100
    </span>
  </div>
</div>
`;

exports[`IndicatorInfo component renders correctly with noTrading=true 1`] = `
<div
  class="indicator-info-wrapper"
  data-testid="indicator-info-wrapper"
>
  <span
    class="text-muted"
  >
    RSI: 
  </span>
  <div
    class="indicator-info__item"
  >
    <span
      class="text-muted"
    >
      RSI 
    </span>
    <span
      class="monospace-text"
    >
      100
    </span>
  </div>
  <span
    class="text-muted"
  >
    No trading
  </span>
</div>
`;
